odoo/models.py

```diff
@@ -4393,6 +4393,9 @@ def _generate_order_by(self, order_spec, query):
     @api.model
     def _flush_search(self, domain, fields=None, order=None, seen=None):
         """ Flush all the fields appearing in `domain`, `fields` and `order`. """
+        if self.env.context.get('skip_flush_search' == self._name):
+            return
```

Comment:
@rco-odoo what do you think? From comment description:

> 
> By default, using `search` method leads to flushing fields that are used in the
> domain. It's totally valid, however it may be unnecessary if we know in advance
> that no new records\updates may change the search result.
> 
> The new option should be used in extreme cases only.